resources/hooks/local/pre-push

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+remote="$1"
+
+while read local_ref local_sha remote_ref remote_sha
+do
+    # Get the branch name
+    local_ref=$(basename $local_ref)
+    break
+done
+    
+# Fetch the GIT diff and format it as command input:
+DIFF=$(git diff --name-only "$remote"/"$local_ref"..HEAD --oneline)
```

Comment:
Maybe it's a better idea to create a new issue in which we can discuss what the pre-push should look like before we implement it. That way we can try to create a good architecture that fits every use-case before implementing 15 different versions?
This PR contains a lot of great findings and was good to determine the scope of the pre-push, but has become a bit messy due to the new issues we've detected during the development of this feature.
What do you think about this idea?